scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_presenter.py

```diff
@@ -0,0 +1,211 @@
+
+    def autoscale_selected_y_axis(self, axis_num):
+        """Autoscales a selected y-axis in the figure using the existing x axis"""
+        self._view.autoscale_selected_y_axis(axis_num)
+
+    def set_axis_limits(self, ax_num, xlims, ylims):
+        """Sets the x and y limits for a specified axis in the figure"""
+        self._view.set_axis_xlimits(ax_num, xlims)
+        self._view.set_axis_ylimits(ax_num, ylims)
+
+    def set_axis_title(self, ax_num, title):
+        """Sets the title for a specified axis in the figure"""
+        self._view.set_title(ax_num, title)
+
+    # Implementation of QuickEdit widget
+    def _update_quickedit_widget(self):
+        self._update_quick_widget_subplots_menu()
+        self._handle_subplot_changed_in_quick_edit_widget()
+
+    def _update_quick_widget_subplots_menu(self):
+        self._options_presenter.clear_subplots()
+        for i in range(self._view.number_of_axes):
+            self._options_presenter.add_subplot(str(i + 1))
+
+    def _handle_subplot_changed_in_quick_edit_widget(self):
+        selected_subplots = self._get_selected_subplots_from_quick_edit_widget()
+        index = selected_subplots[0]
+        xmin, xmax, ymin, ymax = self._view.get_axis_limits(index)
+        self._options_presenter.set_plot_x_range([xmin, xmax])
+        self._options_presenter.set_plot_y_range([ymin, ymax])
+
+    def _handle_xlim_changed_in_quick_edit_options(self, xlims):
+        selected_subplots = self._get_selected_subplots_from_quick_edit_widget()
+        for subplot in selected_subplots:
+            self._view.set_axis_xlimits(subplot, xlims)
+        self._view.redraw_figure()
+
+    def _handle_ylim_changed_in_quick_edit_options(self, ylims):
+        selected_subplots = self._get_selected_subplots_from_quick_edit_widget()
+        for subplot in selected_subplots:
+            self._view.set_axis_ylimits(subplot, ylims)
+        self._view.redraw_figure()
+
+    def _handle_autoscale_y_axes(self):
+        selected_subplots = self._get_selected_subplots_from_quick_edit_widget()
+        if len(selected_subplots) == 1:
+            self.autoscale_selected_y_axis(selected_subplots[0])
+        else:
+            self.autoscale_y_axes()
+
+        self._view.redraw_figure()
+        xmin, xmax, ymin, ymax = self._view.get_axis_limits(selected_subplots[0])
+        self._options_presenter.set_plot_x_range([xmin, xmax])
+        self._options_presenter.set_plot_y_range([ymin, ymax])
+
+    def handle_error_selection_changed(self):
+        plotted_workspaces, _ = self._view.plotted_workspaces_and_indices
+        for workspace_name in plotted_workspaces:
+            self.replot_workspace_with_error_state(workspace_name, self._options_presenter.get_errors())
+
+    def _get_selected_subplots_from_quick_edit_widget(self):
+        subplots = self._options_presenter.get_selection()
+        if len(subplots) > 0:
+            selected_subplots = [ix - 1 for ix in list(map(int, subplots))]
```

Comment:
what is `ix`?

Reply:
Hmm, just index, which is just the integer representation of the subplot which are labeled ("1", "2" ect..).